### ISSUE TITLE
Migration advice sought

### ISSUE THREAD

**Issue by nerdykit** [netPhotoGraphics/netPhotoGraphics#1554]
I'm in the middle of migrating my web server onto a new (faster) computer.  The contents and installation of my nPg gallery are all on an external hard drive and I just have to plug it in to move it over.  The real work is correctly migrating the old mysql database into the new server because that is all new.  I believe that I've managed to do this, I've set up an identical new mysql user and the data in the tables looks good if I examine them with phpmyadmin.  So I'm ready to "switch over" but before I do so, I would really like to be able to test the credentials for the newly created mysql user.  It's been so long since I did the original install of nPg that I've completely forgotten where in the setup process I'm asked for the database prefix, database name and mysql username and password.  So I've no idea where to look to see if I've got all these things done correctly.  I'd hate to corrupt my newly migrated database with a faulty call.  Is there any way to test safely?  ie: utilize what nPg is currently holding as the database credentials against my newly set up database to see if we have connectivity.

**Comment by sbillard** [comment]
The database credentials, etc. are all stored in the npg.cfg.php file in your data folder. If you are replicating the old credentials you should be able to copy the file from the old installation. Then run setup. If you get it wrong nothing will get corrupted, you just will not be able to proceed.

You did not give details of how you migrated the database itself. By far the simplest is to copy a current backup to the new site, create the empty database structure. (If your database server allows it, nPG will create the structure.)  Run setup and when it is complete it will offer to run backup so you can restore the database.

**Comment by nerdykit** [comment]
Thanks, I checked npg.cfg.php and everything seems to match what I've set up on the new server.  I'm not actually required to copy anything except just replicate the database and user to access it.  I did a sql dump from the old database, cp[oed to over, created a new database and just imported the mysql sql file into the new empty database (same name) and then duplicated the user with the same credentials.  I'm just being cautious.
